src/language_features/document_symbol.rs

```diff
@@ -317,13 +317,13 @@ fn next_or_prev_symbol_details<T: Symbol<T> + 'static>(
     for symbol in it {
         let kind = symbol.kind();
```

Comment:
Great fix, thanks!
Since we use the start position as a sorting key, there's a natural asymmetry between searching forward and backward (I believe parents never have lower starting position than their children).

While the old behavior could in theory be useful, it was inconsistent.
This fix means we never skip any children which seems the best behavior (we don't wanna create something as confusing as GNU info ;).

I'll gladly merge as-is but I'll leave a few comments first.

You have a nice reproducer that explains the problem, it would be nice to put that in the commit message (not only on github)

Reply:
@krobelus 
> You have a nice reproducer that explains the problem, it would be nice to put that in the commit message (not only on github)

Done! force pushed commit with reproducer in commit message

> I'll gladly merge as-is but I'll leave a few comments first.

Yes, I would greatly appreciate that. Yes definitely improve the code in subsequent commits as you see fit -- you've already done the ground work in your review comments...

>  also see it as strange that we use < and > instead of something total like < and >= but I haven't found problems)

The omission of the `=` in `>=` is deliberate. IIRC, this was to ensure that we went forward/backward and didn't get "stuck". Basically next/previous always yielded next/previous instead of the same symbol.